specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/stable/2025-06-01/diagnostic.json

```diff
@@ -0,0 +1,307 @@
+        "errordetail": {
+          "type": "string",
+          "description": "The internal error message or exception dump."
+        }
+      },
+      "description": "Inner error details."
+    },
+    "SpotPlacementScoresInput": {
+      "type": "object",
+      "properties": {
+        "desiredLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The desired regions"
+        },
+        "desiredSizes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceSize"
+          },
+          "description": "The desired resource SKUs."
+        },
+        "desiredCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Desired instance count per region/zone based on the scope."
+        },
+        "availabilityZones": {
+          "type": "boolean",
+          "description": "Defines if the scope is zonal or regional."
+        }
+      },
+      "description": "SpotPlacementScores API Input."
+    },
+    "SpotPlacementScoresResponse": {
+      "type": "object",
+      "properties": {
+        "desiredLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The desired regions"
+        },
+        "desiredSizes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceSize"
+          },
+          "description": "The desired resource SKUs."
+        },
+        "desiredCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Desired instance count per region/zone based on the scope."
+        },
+        "availabilityZones": {
+          "type": "boolean",
+          "description": "Defines if the scope is zonal or regional."
+        },
+        "placementScores": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PlacementScore"
+          },
+          "description": "The spot placement scores."
+        }
+      },
+      "description": "SpotPlacementScores API response."
+    },
+    "ResourceSize": {
+      "type": "object",
+      "properties": {
+        "sku": {
+          "type": "string",
+          "description": "The resource's CRP virtual machine SKU size."
+        }
+      },
+      "description": "SpotPlacementRecommender API response."
+    },
+    "PlacementScore": {
+      "type": "object",
+      "properties": {
+        "sku": {
+          "type": "string",
+          "description": "The resource's CRP virtual machine SKU size."
+        },
+        "region": {
+          "type": "string",
+          "description": "The region."
+        },
+        "availabilityZone": {
+          "type": "string",
+          "description": "The availability region."
+        },
+        "score": {
+          "type": "string",
+          "description": "The placement score."
```

Comment:
>"The placement score." [](http://example.com/codeflow?start=25&length=22)

More information about the meaning of this property would be very helpful for clients.

Reply:
Updated